pkg/arvo/app/invite-store.hoon

```diff
@@ -1,184 +1,206 @@
 ::  invite-store [landscape]
-/+  *invite-json, default-agent, dbug
+/-  *invite-store
+/+  res=resource, default-agent, dbug
 |%
 +$  card  card:agent:gall
-::
 +$  versioned-state
-  $%  state-zero
+  $%  state-0
+      state-1
   ==
 ::
-+$  state-zero
-  $:  %0
-      =invites
++$  invitatory-0  (map serial invite-0)
++$  invite-0
+  $:  =ship           ::  ship to subscribe to upon accepting invite
+      app=@tas        ::  app to subscribe to upon accepting invite
+      =path           ::  path to subscribe to upon accepting invite
+      recipient=ship  ::  recipient to receive invite
+      text=cord       ::  text to describe the invite
   ==
+::
++$  state-0  [%0 invites=(map path invitatory-0)]
++$  state-1  [%1 =invites]
 --
 ::
-=|  state-zero
+=|  state-1
 =*  state  -
 %-  agent:dbug
 ^-  agent:gall
-=<
-  |_  bol=bowl:gall
-  +*  this      .
-      inv-core  +>
-      ic        ~(. inv-core bol)
-      def       ~(. (default-agent this %|) bol)
-  ++  on-init   on-init:def
-  ++  on-save   !>(state)
-  ++  on-load
-    |=  old=vase
-    `this(state !<(state-zero old))
+::
+|_  =bowl:gall
++*  this      .
+    def       ~(. (default-agent this %|) bowl)
+::
+++  on-init   on-init:def
+++  on-save   !>(state)
+++  on-load
+  |=  old-vase=vase
+  =/  old  !<(versioned-state old-vase)
+  ?:  ?=(%1 -.old)
+   `this(state old)
+  :-  ~
+  %=  this
+      state
+    :-  %1
+    %-  ~(gas by *^invites)
+    %+  turn  ~(tap by invites.old)
+    |=  [=path =invitatory-0]
+    ^-  [resource invitatory]
+    ?>  ?=([@ @ ~] path)
+    :-  [(slav %p i.path) i.t.path]
+    %-  ~(gas by *invitatory)
+    %+  turn  ~(tap by invitatory-0)
```

Comment:
I'd make this `murn` and use `de-path-soft:res`, just in case we have some bad path formats floating around.

Reply:
done